dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,219 @@ else if( fieldbcontentname.startsWith("date") ){
-            }
+				try {
+					wfActions = APILocator.getWorkflowAPI().findAvailableActions(con, currentUser);
+				} catch (Exception e) {
+					Logger.error(this, "Could not load workflow actions : ", e);
+				}
 
-			JSONArray wfActionMapList = new JSONArray();
+				JSONArray wfActionMapList = new JSONArray();
 
-			for ( WorkflowAction action : wfActions ) {
-				boolean hasPushPublishActionlet = false;
-                if ( action.requiresCheckout() )
-                    continue;
+				for (WorkflowAction action : wfActions) {
+					boolean hasPushPublishActionlet = false;
+					if (action.requiresCheckout())
+						continue;
 
-                JSONObject wfActionMap = new JSONObject();
-                try {
-					wfActionMap.put( "name", action.getName() );
-	                wfActionMap.put( "id", action.getId() );
-	                wfActionMap.put( "icon", action.getIcon() );
-	                wfActionMap.put( "assignable", action.isAssignable() );
-	                wfActionMap.put( "commentable", action.isCommentable() || UtilMethods.isSet( action.getCondition() ) );
-	                wfActionMap.put( "requiresCheckout", action.requiresCheckout() );
-
-	                List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI().findActionClasses(action);
-	                for(WorkflowActionClass actionlet : actionlets){
-	                	if(actionlet.getActionlet() != null 
-	                			&& actionlet.getActionlet().getClass().getCanonicalName().equals(PushPublishActionlet.class.getCanonicalName())){
-	                		hasPushPublishActionlet = true;
-	                	}
-	                }
-	                wfActionMap.put( "hasPushPublishActionlet", hasPushPublishActionlet );
-	                try {
-						wfActionMap.put( "wfActionNameStr", LanguageUtil.get( currentUser, action.getName() ) );
-					} catch (LanguageException e) {
-						Logger.error( this, "Could not load language key : " + action.getName() );
-					}
-	                wfActionMapList.add( wfActionMap );
+					JSONObject wfActionMap = new JSONObject();
+					try {
+						wfActionMap.put("name", action.getName());
+						wfActionMap.put("id", action.getId());
+						wfActionMap.put("icon", action.getIcon());
+						wfActionMap.put("assignable", action.isAssignable());
+						wfActionMap.put("commentable", action.isCommentable() || UtilMethods.isSet(action.getCondition()));
+						wfActionMap.put("requiresCheckout", action.requiresCheckout());
+
+						List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI().findActionClasses(action);
+						for (WorkflowActionClass actionlet : actionlets) {
+							if (actionlet.getActionlet() != null
+									&& actionlet.getActionlet().getClass().getCanonicalName().equals(PushPublishActionlet.class.getCanonicalName())) {
+								hasPushPublishActionlet = true;
+							}
+						}
+						wfActionMap.put("hasPushPublishActionlet", hasPushPublishActionlet);
+						try {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)